nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogic.java

```diff
@@ -226,53 +235,76 @@ public void init(ProcessorInitializationContext context) {
     @OnScheduled
     public void onScheduled(ProcessContext context) {
         super.populatePropertiesByPrefix(context);
+
         dataMovementManager = getDatabaseClient(context).newDataMovementManager();
+
         writeBatcher = dataMovementManager.newWriteBatcher()
             .withJobId(context.getProperty(JOB_ID).getValue())
             .withJobName(context.getProperty(JOB_NAME).getValue())
             .withBatchSize(context.getProperty(BATCH_SIZE).asInteger())
-            .withTemporalCollection(context.getProperty(TEMPORAL_COLLECTION).getValue());
+            .withTemporalCollection(context.getProperty(TEMPORAL_COLLECTION).getValue())
+            .onBatchSuccess(this::onBatchSuccess)
+            .onBatchFailure(this::onBatchFailure);
 
         ServerTransform serverTransform = buildServerTransform(context);
         if (serverTransform != null) {
             writeBatcher.withTransform(serverTransform);
         }
+
         Integer threadCount = context.getProperty(THREAD_COUNT).asInteger();
-        if(threadCount != null) {
+        if (threadCount != null) {
             writeBatcher.withThreadCount(threadCount);
         }
-        this.writeBatcher.onBatchSuccess(writeBatch -> {
-            if (writeBatch.getItems().length > 0) {
-                ProcessSession session = getFlowFileInfoForWriteEvent(writeBatch.getItems()[0]).session;
-                String uriList = Stream.of(writeBatch.getItems()).map((item) -> {
-                    return item.getTargetUri();
-                }).collect(Collectors.joining(","));
-                FlowFile batchFlowFile = session.create();
-                session.putAttribute(batchFlowFile, "URIs", uriList);
-                synchronized(session) {
-                    session.transfer(batchFlowFile, BATCH_SUCCESS);
-                }
-                for(WriteEvent writeEvent : writeBatch.getItems()) {
-                    routeDocumentToRelationship(writeEvent, SUCCESS);
-                }
-            }
-        }).onBatchFailure((writeBatch, throwable) -> {
+
+        dataMovementManager.startJob(writeBatcher);
+    }
+
+    protected void onBatchSuccess(WriteBatch writeBatch) {
+        if (writeBatch.getItems().length > 0) {
+            getLogger().info("Inserted {} documents from batch number {}",
```

Comment:
This is the actual logging fix - logging it after success and including the correct number of documents and the batch number too (which I find helpful).